tracer/src/Datadog.Tracer.Native/clr_helpers.cpp

```diff
@@ -1004,4 +1004,41 @@ HRESULT FunctionLocalSignature::TryParse(PCCOR_SIGNATURE pbBase, unsigned len, s
     return S_OK;
 }
 
+HRESULT HasAsyncStateMachineAttribute(const ComPtr<IMetaDataImport2>& metadataImport, const mdMethodDef methodDefToken)
+{
+    const void* ppData = nullptr;
+    ULONG pcbData = 0;
+    const auto hasAsyncStateMachineAttrHr = metadataImport->GetCustomAttributeByName(
+        methodDefToken, WStr("System.Runtime.CompilerServices.AsyncStateMachineAttribute"), &ppData, &pcbData);
+
+    if (hasAsyncStateMachineAttrHr == S_OK)
+    {
+        if (pcbData > 0)
+        {
+            return S_OK;
+        }
+        return S_FALSE;
+    }
+
+    return hasAsyncStateMachineAttrHr;
```

Comment:
```suggestion
    IfFailRet(hasAsyncStateMachineAttrHr);
    return pcbData > 0 ? S_OK : S_FALSE;
```

Reply:
No biggy, feel free to resolve this suggestion.

Reply:
Fixed in 1fa418a83881f1af240463893db4ee34fb543c45